tools/generate_pv_yaml_rest.sh

```diff
@@ -0,0 +1,328 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+usage() {
+  echo "Usage: $0
+                -f|--filesystem <Name of Volume's Source Filesystem>
+                -l|--linkpath <full Path of Volume in Primary Filesystem>
+                -F|--fileset <name of source fileset>
+                -s|--size <size in GB>
+                -u|--username <Username of spectrum scale GUI user account.>
+                -t|--password <Password of spectrum scale GUI user account.>
+                -r|--route <Route host name used to route traffic to the spectrum scale GUI service.>
+                [-p|--pvname <name for pv>]
+                [-c|--storageclass <StorageClass for pv>]
+                [-a|--accessmode <AccessMode for pv>]
+                [-h|--help] " 1>&2
+  exit 1
+}
+
+fullUsage() {
+  echo "Usage: $0
+                -f|--filesystem <Name of Volume's Source Filesystem>
+                -l|--linkpath <full Path of Volume in Primary Filesystem>
+                -F|--fileset <name of source fileset>
+                -s|--size <size in GB>
+                -u|--username <Username of spectrum scale GUI user account.>
+                -t|--password <Password of spectrum scale GUI user account.>
+                -r|--route <Route host name used to route traffic to the spectrum scale GUI service.>
+                [-p|--pvname <name for pv>]
+                [-c|--storageclass <StorageClass for pv>]
+                [-a|--accessmode <AccessMode for pv>]
+                [-h|--help]
+
+
+Example 1: Single Fileystem
+	In this setup there is only one fileystem 'gpfs0' and directory from the same fileystem is being used as volume.
+
+	$0 --filesystem gpfs0 --linkpath /ibm/gpfs0/fileset1/.volumes/staticpv --size 10 --pvname mystaticpv --username admin --password password --route ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.hci-cluster.cp.fyre.ibm.com
+
+
+Example 2: Two or More Filesystem
+	In this setup there are two filesystems 'gpfs0' and 'gpfs1'. gpfs0 is configured as primary fileystem in Spectrum-scale-csi setup. User want to create volume from the directory present in the gpfs1 filesystem. Say the directory in the gpfs1 is /ibm/gpfs1/dir1. As a first step user will create softlink  /ibm/gpfs1/dir1 --> /ibm/gpfs0/fileset1/.volumes/staticpv1 and then run following command to generate the pv.yaml.
+
+	$0 --filesystem gpfs1 --linkpath /ibm/gpfs0/fileset1/.volumes/staticpv1 --size 10 --pvname mystaticpv1 --username admin --password password --route ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.hci-cluster.cp.fyre.ibm.com
+
+Example 3: Fileset based volume
+	This example shows how to create a volume from a fileset 'fileset1' within the filesyetem 'gpfs0'.
+
+	$0 --filesystem gpfs0 --fileset fileset1 --size 10 --pvname mystaticpv --username admin --password password --route ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.hci-cluster.cp.fyre.ibm.com
```

Comment:
will this script work even in case of CNSA with remote storage access or it is designed only for fusion?

Reply:
The script works in case of CNSA with remote storage access as well.

Example:

Remote filesystem mounted in /mnt path
```
remotefs       gpfs     184G  2.6G  181G   2% /mnt/remotefs
```
Script command:
```
# bash generate_pv_yaml_rest.sh --filesystem remotefs --linkpath /mnt/remotefs/remotefstest1 --username admin --password password  --size 10 --pvname mystaticpv --guihost ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.ece-wip-test.cp.fyre.ibm.com
```

Output
```
FSETNAME=
INFO: volumeHandle: 6047684167693110895;0A0B70C5:5FE1909A;path=/gpfs/convulse1.fyre.ibm.com/remotefstest1
INFO: Successfully created mystaticpv.yaml
INFO: Successfully created pvc-mystaticpv.yaml
```
Tested on sample pod, was able to mount the filesystem using the generated PV and PVC.